lms/static/js/ccx/schedule.js

```diff
@@ -251,9 +269,27 @@ var edx = edx || {};
 	    }
 	},
 
+	valid_dates: function(start, due) {
+          var errorMessage;
+          // Start date is compulsory and due date is optional.
+          if (_.isEmpty(start) && !_.isEmpty(due)) {
+            errorMessage = gettext("Please enter valid start date and time.");
+          } else if (!_.isEmpty(start) && !_.isEmpty(due)) {
+            var requirejs = require || RequireJS.require;
```

Comment:
If you're concerned that the global `require` function might not be defined, then you should check like this:
```js
var requirejs = window.require || RequireJS.require;
```
If `require` is actually not defined, trying to access it directly will cause a ReferenceError. However, `window` is always defined in the browser, and global variables are assigned to the `window` object. If `require` is defined as a global variable, `window.require` will return it. If it is not defined as a global variable, `window.require` will return `undefined`, rather than throwing a ReferenceError.

It would be better to just wrap this entire file in a `require()` function call, but that change seems out of scope for this pull request, so this solution is fine.